src/robusta/core/sinks/robusta/dal/model_conversion.py

```diff
@@ -86,6 +86,33 @@ def get_empty_file_object(block: EmptyFileBlock):
             "data": "",
         }
 
+    @staticmethod
+    def append_to_structured_data_tool_calls(tool_calls: List[ToolCallResult], structured_data) -> None:
+        for tool_call in tool_calls:
+            file_block = FileBlock(f"{tool_call.description}.txt", tool_call.result.encode())
+            file_block.zip()
+            data_obj = ModelConversion.get_file_object(file_block)
+            data_obj["metadata"] = {"description": tool_call.description, "tool_name": tool_call.tool_name}
+            structured_data.append(data_obj)
+
+    @staticmethod
+    def add_ai_chat_data(structured_data: List[Dict], block: HolmesChatResultsBlock):
+        structured_data.append(
+            {
+                "type": "markdown",
+                "metadata": {"type": "ai_investigation_result", "createdAt": datetime_to_db_str(datetime.now())},
+                "data": Transformer.to_github_markdown(block.holmes_result.analysis),
+            }
+        )
+        ModelConversion.append_to_structured_data_tool_calls(block.holmes_result.tool_calls, structured_data)
+
+        conversation_history_block = FileBlock(
```

Comment:
Why is this a file block?
Wouldn't a json block be better here?

Reply:
It makes sense to me to use FileBlock as conversation_history can become pretty huge, and a few lines upper we use FileBlock to compress tool's output (just as a reminder, conversation_history includes tools and all messages)